plugin/path_reserves.go

```diff
@@ -151,6 +159,12 @@ func (b *backend) operationReserveUpdate(ctx context.Context, req *logical.Reque
 	}
 	lendingPeriod := time.Duration(lendingPeriodRaw.(int)) * time.Second
 
+	disableCheckInEnforcementRaw, enforcementSent := fieldData.GetOk("disable_check_in_enforcement")
+	if !enforcementSent {
```

Comment:
`GetOk` will return the default `false` here, is that correct? Assuming so we can do away with the check here, I think?

Reply:
I thought that originally, but found that wasn't the case. If an optional field isn't sent by the caller, and you use `Get`, it does indeed default the value. _But_ if an optional field isn't sent by the caller, and you use `GetOk`, it returns `nil, false`. So then if you try to cast nil to a strong type using `raw.(string)` (supposing that raw is nil), you get a panic.

That's why I ended up doing it that way, because I got panics in testing.

Reply:
Today I learned! I was expecting `GetOk` to return `<default>, nil` if it wasn't set, but clearly it does not behave that way 🤔